flex_config/__init__.py

```diff
@@ -22,21 +22,27 @@ def _merge_sources(dest: Dict[str, Any], source: ConfigSource) -> Dict[str, Any]
     return dest
 
 
-def _compile_sources(sources: Union[Sequence[ConfigSource], ConfigSource]) -> Dict[str, Any]:
+def _compile_sources(
+    sources: Union[Sequence[Union[ConfigSource, ConfigSchema]], ConfigSource, ConfigSchema]
```

Comment:
I would store the `Union[ConfigSource, ConfigSchema]` as a var called `SourceTypes` or something to make this less busy.